x/dsmr/storage.go

```diff
@@ -228,20 +238,24 @@ func (s *ChunkStorage[T]) VerifyRemoteChunk(c Chunk[T]) (*warp.BitSetSignature,
 	return nil, nil
 }
 
+// putVerifiedChunk assumes that the given chunk is gurenteed not to surpass the producer's rate limit.
+// The rate limit is being checked via a call to CheckRateLimit from BuildChunk as well as by VerifyRemoteChunk.
```

Comment:
Could we make sure this comment is up to date? This is called to self-regulate within `BuildChunk` and it is called in the p2p handler `ChunkSignatureRequestVerifier` to rate limit peers, but this is not called in `VerifyRemoteChunk` atm.

Also, with the current code, it's not safe to call this within `VerifyRemoteChunk` because we call `VerifyRemoteChunk` within `Accept`. This means that if a peer consumed more bandwidth, this could fail within accept (which would fatal).

Reply:
updated.